File: pec-observation-app/services/ms-011-knowledge/app/domain/knowledge.py
from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import Optional


class DocumentType(str, Enum):
    SEDUC_POLICY      = "seduc_policy"
    CURRICULUM_GUIDE  = "curriculum_guide"
    EVALUATION_RUBRIC = "evaluation_rubric"
    FEEDBACK_EXAMPLE  = "feedback_example"
    OTHER             = "other"


class StyleTone(str, Enum):
    FORMAL       = "formal"
    CONSTRUCTIVE = "constructive"
    DIRECT       = "direct"
    SUPPORTIVE   = "supportive"


@dataclass
class DocumentChunk:
    chunk_index: int
    text: str
    char_start: int
    char_end: int


@dataclass
class KnowledgeDocument:
    id: str
    title: str
    document_type: DocumentType
    source_filename: str
    full_text: str
    chunks: list[DocumentChunk]
    chunk_count: int
    uploaded_by: str
    is_active: bool
    created_at: datetime
    description: Optional[str] = None


@dataclass
class FeedbackStyle:
    id: str
    name: str
    description: str
    tone: StyleTone
    template_prompt: str          # injected into LLM system prompt
    example_strengths: list[str]
    example_improvements: list[str]
    is_active: bool
    is_default: bool
    created_at: datetime


@dataclass
class ContextResult:
    """Relevant chunks returned to MS-006 for a given observation."""
    chunks: list[str]
    document_titles: list[str]
    active_style: Optional[FeedbackStyle]

File: pec-observation-app/web-app/src/domain/entities/Knowledge.ts
export type DocumentType =
  | 'seduc_policy'
  | 'curriculum_guide'
  | 'evaluation_rubric'
  | 'feedback_example'
  | 'other'

export const DOCUMENT_TYPE_LABELS: Record<DocumentType, string> = {
  seduc_policy:      'Política SEDUC',
  curriculum_guide:  'Guia Curricular',
  evaluation_rubric: 'Rubrica de Avaliação',
  feedback_example:  'Exemplo de Feedback',
  other:             'Outro',
}

export interface KnowledgeDocument {
  id: string
  title: string
  documentType: DocumentType
  sourceFilename: string
  description?: string
  chunkCount: number
  uploadedBy: string
  isActive: boolean
  createdAt: string
}

export type StyleTone = 'formal' | 'constructive' | 'direct' | 'supportive'

export const STYLE_TONE_LABELS: Record<StyleTone, string> = {
  formal:       'Formal',
  constructive: 'Construtivo',
  direct:       'Direto',
  supportive:   'Encorajador',
}

export interface FeedbackStyle {
  id: string
  name: string
  description: string
  tone: StyleTone
  templatePrompt: string
  exampleStrengths: string[]
  exampleImprovements: string[]
  isActive: boolean
  isDefault: boolean
  createdAt: string
}
